### PR TITLE
Je/python3

### PR DESCRIPTION
**Pull request by emord**
open for tests

### CODE REVIEW

**Comment by emord** [comment]
@nickpell @millerdev This should be good to go. The failure was just the python 3 job and I removed that

File: dimagi/utils/csv.py [hunk 1]
```diff
@@ -19,7 +19,11 @@ def __init__(self, f, encoding):
     def __iter__(self):
         return self
 
+    def __next__(self):
+        return self.reader.next().encode("utf-8")
```
Comment (millerdev) [inline dimagi/utils/csv.py]: should this be `next(self.reader).encode("utf-8")`

File: dimagi/utils/csv.py [hunk 2]
```diff
@@ -19,7 +19,11 @@ def __init__(self, f, encoding):
...
+
     def next(self):
```
Comment (millerdev) [inline dimagi/utils/csv.py]: Could change this to
```py
next = __next__  # For Py2 compatibility
```
and avoid some duplication

File: dimagi/utils/csv.py [hunk 3]
```diff
@@ -32,13 +36,18 @@ def __init__(self, f, dialect=csv.excel, encoding="utf-8", **kwds):
...
-    def next(self):
-        row = self.reader.next()
+    def __next__(self):
+        row = next(self.reader)
         return [unicode(s, "utf-8") for s in row]
 
     def __iter__(self):
         return self
 
+    def next(self):
```
Comment (millerdev) [inline dimagi/utils/csv.py]: Same as above comment
```py
next = __next__  # For Py2 compatibility
```

File: dimagi/utils/tests/cache_tests.py [hunk 4]
```diff
@@ -15,8 +15,8 @@ def get(self, key):
         return self.cache_dict.get(key, None)
 
     def keys(self, pattern):
-        all_keys = self.cache_dict.keys()
-        filtered = filter(lambda x: x.startswith(pattern[:-1]), all_keys)
+        all_keys = list(self.cache_dict)
```
Comment (millerdev) [inline dimagi/utils/tests/cache_tests.py]: `list()` not needed

File: dimagi/utils/gitinfo.py [hunk 5]
```diff
@@ -42,9 +42,9 @@ def sub_git_cmd(git_dir, args):
             ] + args,
             stdout=PIPE, stderr=PIPE
         )
-    except OSError, e:
+    except OSError as e:
         # Is git missing ?
-        if type(e) is OSError and e.errno == 2:
+        if isinstance(e, OSError) and e.errno == 2:
```
Comment (nickpell) [inline dimagi/utils/gitinfo.py]: can we remove the ```isinstance``` check?

**Pull request merged**